tests/unit/config/test_esgf_pyclient.py

```diff
@@ -13,16 +13,24 @@
         'bootstrap': True,
     },
     'search_connection': {
-        'url':
-        'http://esgf-node.llnl.gov/esg-search',
+        'urls': [
+            'https://esgf-index1.ceda.ac.uk/esg-search',
+            'https://esgf-node.llnl.gov/esg-search',
+            'https://esgf-data.dkrz.de/esg-search',
+            'https://esgf-node.ipsl.upmc.fr/esg-search',
+            'https://esg-dn1.nsc.liu.se/esg-search',
+            'https://esgf.nci.org.au/esg-search',
+            'https://esgf.nccs.nasa.gov/esg-search',
+            'https://esgdata.gfdl.noaa.gov/esg-search',
+        ],
```

Comment:
cheers @bouweandela - I think it looks good, but I've not tested it, I trust you did, and probably best if you turn on the GA tests and rerun the tests a few times (if you expect a much less frequent failure due to node not being reachable). I do think we should put this list of nodes someplace centrally and just load it from there both for the actual code and for the testing environment, I have seen this list repeated at least three times in different places, and that's prone to introducing issues if we change one node somewhere but not everywhere. What you reckon?

Reply:
Unfortunately, I have not been able to test this in real life either, because you need an index node that is timing out for that and I don't think it's that common.

The risk of introducing an error because of the duplication is not that big, the worst thing that can happen is that the documentation goes out of date. The unit test ensures that the default configuration and test code are identical. I agree that this may not be the most elegant unit test, but reading the list from the same source in production and test code is asking for trouble.

Reply:
> Unfortunately, I have not been able to test this in real life either, because you need an index node that is timing out for that and I don't think it's that common.

Yeah that's why I was thinking we might pick that up from running a bunch of GA tests - but it's prob too much hassle - we'll have to just wait and see on nightly/PR merge tests I reckon